assets/animations.js

```diff
@@ -0,0 +1,25 @@
+function onIntersection(elements, observer) {
+  elements.forEach((element, index) => {
+    if (element.isIntersecting) {
+      const elementTarget = element.target;
+      elementTarget.classList.add('animation__trigger--active');
+      elementTarget.setAttribute('style', `--animation-order: ${index};`);
+
+      observer.unobserve(elementTarget);
+    }
+  })
+}
+
+function initializeAnimationTrigger() {
```

Comment:
I think we should avoid using the word "animation" in this module to keep a good separation of concerns. To me this observer is just about watching for when an element is scrolled into view, and has no opinion on what to do when that happens.

Reply:
That's a fair comment, but it is the observer _for_ elements with the class name `animation__trigger`, so there's already a reference to that in the function. What would you suggest?

Reply:
Yeah I prefer `initializeScrollAnimationTrigger` - we could imagine using the intersection observer for other non-animation things in the future like analytics and lazy loading. Keeping it generic will make it more flexible.